### PR TITLE
refactor: Consolidate end to end normalization code

### PR DESCRIPTION
**Pull request by alamb**
I recommend viewing with whitespace blind diff; https://github.com/influxdata/influxdb_iox/pull/6848/files?w=1

# Rationale

In an effort to try and help the end to end testing harness be easier to work with given we have now have many tests that use it, I am trying to split it up more. 

# Changes
1. Move all end to end test normalization and configuration code into the normalize module
2. Move regex construction to `Lazy` variables, and add more comments

This code could certainly use more cleanup, but I think this at least helps

### CODE REVIEW

File: test_helpers_end_to_end/src/snapshot_comparison.rs [hunk 1]
```diff
@@ -98,19 +96,7 @@ pub async fn run(
...
-            output.push("-- Results After Normalizing UUIDs".into())
-        }
-        if q.normalized_metrics() {
-            output.push("-- Results After Normalizing Metrics".into())
-        }
-        if q.normalized_filters() {
-            output.push("-- Results After Normalizing Filters".into())
-        }
-
+        q.add_decription(&mut output);
```
Comment (alamb) [inline test_helpers_end_to_end/src/snapshot_comparison.rs:99]: code was moved

File: test_helpers_end_to_end/src/snapshot_comparison/normalization.rs [hunk 2]
```diff
@@ -1,9 +1,28 @@
-use crate::snapshot_comparison::queries::Query;
+use arrow::record_batch::RecordBatch;
+use arrow_util::{display::pretty_format_batches, test_util::sort_record_batch};
 use once_cell::sync::Lazy;
 use regex::{Captures, Regex};
 use std::{borrow::Cow, collections::HashMap};
 use uuid::Uuid;
 
+/// Match the parquet UUID
```
Comment (alamb) [inline test_helpers_end_to_end/src/snapshot_comparison/normalization.rs:8]: extracted these regular expressions into `Lazy` variables and added comments about what they do

File: test_helpers_end_to_end/src/snapshot_comparison/normalization.rs [hunk 3]
```diff
@@ -22,93 +41,159 @@ static REGEX_LINESEP: Lazy<Regex> = Lazy::new(|| Regex::new(r#"[+-]{6,}"#).expec
...
+                })
+                .collect();
+        }
+
+        current_results
     }
 
-    current_results
+    /// Adds information on what normalizations were applied to the input
+    pub fn add_decription(&self, output: &mut Vec<String>) {
```
Comment (carols10cents) [inline test_helpers_end_to_end/src/snapshot_comparison/normalization.rs:185]:
```suggestion
    pub fn add_description(&self, output: &mut Vec<String>) {
```

File: test_helpers_end_to_end/src/snapshot_comparison/queries.rs [hunk 4]
```diff
@@ -27,49 +17,49 @@ impl Query {
...
+    pub fn normalize_results(&self, results: Vec<RecordBatch>) -> Vec<String> {
+        self.normalizer.normalize_results(results)
     }
 
-    /// Use normalized filter plans
-    pub fn normalized_filters(&self) -> bool {
-        self.normalized_filters
+    /// Adds information on what normalizations were applied to the input
+    pub fn add_decription(&self, output: &mut Vec<String>) {
+        self.normalizer.add_decription(output)
```
Comment (carols10cents) [inline test_helpers_end_to_end/src/snapshot_comparison/queries.rs:56]:
```suggestion
    pub fn add_description(&self, output: &mut Vec<String>) {
        self.normalizer.add_description(output)
```

File: test_helpers_end_to_end/src/snapshot_comparison.rs (hunk 1 above)
Comment (carols10cents) [inline test_helpers_end_to_end/src/snapshot_comparison.rs:99]: I think `add_decription` is supposed to be `add_description`? I'm going through and adding suggestions so you can click them to commit them if that's correct--


```suggestion
        q.add_description(&mut output);
```

File: test_helpers_end_to_end/src/snapshot_comparison/queries.rs [hunk 5]
```diff
@@ -125,54 +99,57 @@ impl TestQueries {
...
-                        }
-                        "metrics" => {
-                            builder.normalize_metrics();
-                        }
-                        "filters" => {
-                            builder.normalize_filters();
+
+        let mut builder = lines
+            .into_iter()
+            .fold(QueryBuilder::new(), |mut builder, line| {
```
Comment (carols10cents) [inline test_helpers_end_to_end/src/snapshot_comparison/queries.rs:105]: 👍🏻 on the `fold` :)

**Review by carols10cents — APPROVED** [review]
Just a question on the spelling of a method name that I trust you to resolve, possibly with the suggestions I've made :)

File: test_helpers_end_to_end/src/snapshot_comparison.rs (hunk 1 above)
Comment (alamb) [reply test_helpers_end_to_end/src/snapshot_comparison.rs:99]: :facepa

**Comment by alamb** [comment]
rebased to fix semantic commit message

**Pull request merged**